dfu_target SUIT cache implementation and unit tests. (#17281)
nrfconnect/sdk-nrf: doc/nrf/libraries/dfu/dfu_multi_image.rst

|               ``SB_CONFIG_DFU_MULTI_IMAGE_PACKAGE_NET``           | Include network core image update.     |
+-------------------------------------------------------------------+----------------------------------------+
|               ``SB_CONFIG_DFU_MULTI_IMAGE_PACKAGE_MCUBOOT``       | Include MCUboot update.                |
+-------------------------------------------------------------------+----------------------------------------+
|               ``SB_CONFIG_DFU_MULTI_IMAGE_PACKAGE_WIFI_FW_PATCH`` | Include nRF700x WiFi firmware patches. |
+-------------------------------------------------------------------+----------------------------------------+
|               ``SB_CONFIG_SUIT_MULTI_IMAGE_PACKAGE_BUILD``        | Include SUIT envelope and cache images.|
+-------------------------------------------------------------------+----------------------------------------+

.. _lib_dfu_multi_image_suit_multi_image_package:

SUIT multi-image package
========================

The DFU multi-image library supports building a SUIT multi-image package that includes a SUIT envelope and cache images.

> The SUIT envelope is always included in the package as image 0, while SUIT cache images are included as subsequent images starting from image 2.

Reviewer: ```suggestion
The SUIT envelope is always included in the package as image 0, while SUIT cache images are included as subsequent images starting from the image 2.
```

The SUIT multi-image processing requires the SUIT system to support cache processing.

> To enable it set one of the following Kconfig options to ``y``:

Reviewer: ```suggestion
To enable it, set one of the following Kconfig options to ``y``:
```

* :kconfig:option:`CONFIG_SUIT_DFU_CANDIDATE_PROCESSING_FULL`
* :kconfig:option:`SUIT_DFU_CANDIDATE_PROCESSING_PUSH_TO_CACHE`

The build system merges all application images into a single `dfu_cache_partition_1.bin` partition file and places its content into multi-image image 2.
This allows all application images to be stored in a single DFU multi-image, as they will be processed by SUIT.

The :kconfig:option:`SB_CONFIG_SUIT_MULTI_IMAGE_PACKAGE_BUILD` Kconfig option enables building the SUIT multi-image package.
As a result, the multi-image package will contain:

* Image 0:
   - SUIT envelope that contains manifests only.

* Image 2:
   - Application core image.
   - Network core image, if applicable.
   - Additional images, if applicable.

You can add more data to be processed by SUIT to the following images starting from image 3.

> This operation will require an additional `dfu_cache_partition_X` definition, where X is the image number minus 1.

Reviewer: ```suggestion
This operation will require an additional :file:`dfu_cache_partition_X` file and definition, where ``X`` is the image number minus 1.
```

> So for image 3, `dfu_cache_partition_2` is required.

Reviewer: ```suggestion
So for the image 3, you would need :file:`dfu_cache_partition_2`.
```
